tests/components/google/conftest.py

```diff
@@ -304,3 +305,9 @@ async def _setup_func() -> bool:
         return result
 
     return _setup_func
+
+
+@pytest.fixture(autouse=True)
+async def setup_application_credentials(hass: HomeAssistant) -> None:
```

Comment:
Why is this needed? Isn't this done when setting up the Google integration as a dependency?

Reply:
My impression is that config flows are invoked before component setup when there is no domain in the configuration. Let me know if I am mistaken or that is not supposed to be the case due to a setup bug.

Reply:
So this is needed for the config flow tests that don't use the `component_setup` fixture?

Reply:
Yes that is right. Perhaps, then, we only include this in those cases, but it all feels a bit loose to be so selective.

Perhaps invoking a config flow of a component should load it's dependencies or setup? I hit this in nest also when trying to move away from yaml, so surprised it's not more common.

Reply:
I double checked this and my recollection was incorrect: It was needed for loading credentials for a config entry before the component is loaded. I commented this out and it was only needed really in one place so i just moved it there.